odoo/sql_db.py

```diff
@@ -63,7 +63,8 @@ def adapt_string(adapted):
 def flush_env(cr):
     """ Retrieve and flush an environment corresponding to the given cursor """
     for env in list(Environment.envs):
-        if env.cr is cr:
+        # don't flush() on another cursor or with a RequestUID
+        if env.cr is cr and isinstance(env.uid, int):
```

Comment:
The problem is that by testing for `int` instead of `RequestUID`, cases where `env.uid == None` are excluded as well. So `auth='none'` routes aren't flushed anymore, and commit an empty transaction unless something else flushed by chance at some point.

Reply:
```diff
- 	   if env.cr is cr and isinstance(env.uid, int):
+          if env.cr is cr and (isinstance(env.uid, int) or env.uid is None):
```

Reply:
Cfr fix in #48685